src/mlpack/tests/ann_test_tools.hpp

```diff
@@ -98,6 +98,65 @@ double JacobianTest(ModuleType& module,
   return arma::max(arma::max(arma::abs(jacobianA - jacobianB)));
 }
 
+// Custom Jacobian Test where we get the input from outside of this function
+// unlike the original Jacobian Test where input is generated inside that
+// funcion.
```

Comment:
This was the problem why initially Jacobian Test failed for the case `query = key = value`.